### PR TITLE
Hive: removed Iceberg props should be removed from HMS table props too

### PR DESCRIPTION
**Pull request by marton-bod**
Addressing https://github.com/apache/iceberg/issues/2249

### CODE REVIEW

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java [hunk 1]
```diff
@@ -186,7 +188,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
             baseMetadataLocation, metadataLocation, database, tableName);
       }
 
-      setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), hiveEngineEnabled);
+      // get Iceberg props that have been removed
+      List<String> removedProps = Collections.emptyList();
```
Comment (pvary) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:192]: Could this be a set?

Comment (marton-bod) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java]: Sure, changed it to a set.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java [hunk 2]
```diff
@@ -258,7 +268,7 @@ private Table newHmsTable() {
   }
 
   private void setHmsTableParameters(String newMetadataLocation, Table tbl, Map<String, String> icebergTableProps,
-      boolean hiveEngineEnabled) {
+      Set<String> obsoleteProps, boolean hiveEngineEnabled) {
```
Comment (aokolnychyi) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:271]: nit: I know it is not very consistent throughout the codebase but I think the following formatting would be more aligned with the rest of the code:

image

**Review by aokolnychyi — APPROVED** [review]
LGTM. Thanks, @marton-bod!

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java (hunk 2 above)
Comment (marton-bod) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java]: Makes sense, adjusted it

**Comment by marton-bod** [comment]
Thanks for the review @aokolnychyi , @pvary !

**Pull request merged**